testament/categories.nim

```diff
@@ -479,16 +479,6 @@ proc makeSupTest(test, options: string, cat: Category): TTest =
   result.options = options
   result.startTime = epochTime()
 
-proc actionRetry(maxRetry: int, backoffDuration: float, action: proc: bool): bool =
```

Comment:
@Clyybber why did you remove this? this was there for a good reason, reducing spurious git clone failures which used to happen a lot before this was added

Reply:
ping @Clyybber , I'm now getting 503/443 errors again after your removal of `actionRetry` which was introduced for this very purpose in https://github.com/nim-lang/Nim/pull/14547

eg: in https://github.com/nim-lang/Nim/pull/15765/checks?check_run_id=1323632253 i just got:

```
Installing recaptcha@>= 1.0.3
  Downloading euantorano/recaptcha.nim using git
  download.nim(93)         getTagsListRemote
  Error: unhandled exception: Unable to query remote tags for euantorano/recaptcha.nim. Git returned: fatal: unable to access 'euantorano/recaptcha.nim': Failed to connect to github.com port 443: Connection refused
   [OSError]
```

Reply:
ping @Clyybber this keeps happening

Reply:
Yeah...

Reply:
Sorry, this slipped my notifications. PR incoming